File: bankomat/typescript/src/models/user-account.ts
import { AssignedCards } from "./assigned-cards";
import { Operation } from "./operation";
import { UserData } from "./user.data";

export class UserAccount {
    userId: number;
    firstName: string;
    lastName: string;
    assignedCards: AssignedCards[]

    constructor(data: UserData) {
        this.userId = data.id;
        this.firstName = data.first_name;
        this.lastName = data.last_name;
        this.assignedCards = data.assignedCards.map((item) => new AssignedCards(item));
    }
}

File: bankomat/python/UserAccount.py
import functools

from Operation import Operation
from AssignedCards import AssignedCards

class UserAccount():

    def __init__(self, data):

        self.id = data["id"]
        self.name = data["first_name"]
        self.surname = data["last_name"]
        self.assignedCards = list(map(
            lambda item: AssignedCards(item), data["assignedCards"]))
        self.operations = list(map(lambda item: Operation(item), data["operations"]))

    def getBalance(self):
        return functools.reduce(lambda prev, curr: prev + curr.moneyNum, self.operations, 0)
